code-studio/src/js/components/GridEditor.jsx

```diff
@@ -101,39 +111,86 @@ var GridEditor = React.createClass({
     // Both of those seem a bit unnecessary, so for now this hack will
     // remain.
     var cells = this.state.cells;
-    var newCell = this.getCellClass().deserialize(newSerializedCell);
-    if (row !== undefined && col !== undefined) {
-      cells[row][col] = newCell;
-    }
+    newCells.forEach(function (_, i) {
+      _.forEach(function (cell, j) {
```

Comment:
Nit: When I see `_` as a parameter name I usually assume that it's unused/irrelevant, but you use it here.  Maybe `newRow` would be appropriate?